src/Orleans/Core/IInternalGrainFactory.cs

```diff
@@ -47,5 +47,21 @@ TGrainInterface GetSystemTarget<TGrainInterface>(GrainId grainId, SiloAddress de
         /// A reference to <paramref name="grain"/> which implements <paramref name="interfaceType"/>.
         /// </returns>
         object Cast(IAddressable grain, Type interfaceType);
+
+        /// <summary>
+        /// Gets a reference to the grain with the provided id.
+        /// </summary>
+        /// <typeparam name="TGrainInterface">The grain reference interface type.</typeparam>
+        /// <param name="grainId">The grain id.</param>
+        /// <returns>A reference to the grain with the provided id.</returns>
+        TGrainInterface GetGrain<TGrainInterface>(GrainId grainId);
```

Comment:
I'm fine with that, the issue is that those things are all 'grain references' in our system, so the logical name might be 'GetGrainReference', but that's no better than GetGrain. Still, I'm fine calling it `GetAddressableReference` if you want to make it explicit - it's just that the distinction is a little shaky